internal/execution/request.go

```diff
@@ -0,0 +1,74 @@
+package execution
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/zitadel/logging"
+
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/query"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+type ContextInfoRequest struct {
+	FullMethod string      `json:"fullMethod,omitempty"`
+	InstanceID string      `json:"instanceID,omitempty"`
+	OrgID      string      `json:"orgID,omitempty"`
+	ProjectID  string      `json:"projectID,omitempty"`
+	UserID     string      `json:"userID,omitempty"`
+	Request    interface{} `json:"request,omitempty"`
+}
+
+func CallTargetsRequest(ctx context.Context,
+	targets []*query.Target,
+	info *ContextInfoRequest,
+) (interface{}, error) {
+	ret := info.Request
+	for _, target := range targets {
+		if target.Async {
```

Comment:
As far as I remember we had a discussion about removing `target.Async` and adding an additional type `REST Async Call` this would simplify the objects and would simplify the API in general

Reply:
As it makes the code more complex than it would be without that field i would change that now